Dockerfile

```diff
@@ -1,6 +1,26 @@
-FROM node:18-alpine
-WORKDIR /app
-EXPOSE  3000
-COPY . .
+FROM node:20-alpine AS BUILD_IMAGE
+
+RUN mkdir -p /usr/app/
+WORKDIR /usr/app/
+
+COPY ./ ./
```

Comment:
This can be further optimized by copying just the package.json file -> running npm install -> then copying remaining files. This will work well with caching.

Reply:
Thanks, @Harsh2509. Got it. That way, if any dependency changes, it will run npm install; otherwise, it won't. If you want, you can create a PR, @Harsh2509